pegasus/helpers/hourofcode_helpers.rb

```diff
@@ -29,12 +29,40 @@ def trans_load_i18n
 end
 TRANS_I18N = trans_load_i18n
 
-def hoc_s(id)
+# For internal use by hoc_s and hoc_s_by_locale.
+# Looks up a string in the hourofcode translations using the provided two-letter
+# language.
+def hoc_s_by_language(id, language)
   id = id.to_s
-
   return TRANS_I18N['en-US'][id] if request.site == 'translate.hourofcode.com'
+  HOC_I18N[language][id] || HOC_I18N['en'][id]
+end
+
+# Called by pages on hourofcode.com, which have two-letter language in @language.
+def hoc_s(id)
+  hoc_s_by_language(id, @language)
+end
+
+# Called by pages on code.org/csedweek.org, which have XX-XX locale in
+# request.locale.
+def hoc_s_by_locale(id)
+  locale = request.locale[0..1]
+
+  case locale
+  when 'en-GB'
```

Comment:
I wonder, is it possible for us to reference the cdo-languages gsheet where I've outlined these exceptions? Because for future ones I would have to update the gsheet and here.